test/protocol/MetaTransactionsHandlerTest.js

```diff
@@ -637,8 +716,85 @@ describe("IBosonMetaTransactionsHandler", function () {
+              [keccak256(toUtf8Bytes(message.functionName))],
+              false
+            );
+
+            // Prepare the function signature for the facet function.
+            functionSignature = accountHandler.interface.encodeFunctionData("createSeller", [
+              seller,
+              emptyAuthToken,
+              voucherInitValues,
+            ]);
+
+            // Prepare the message
+            message.functionSignature = functionSignature;
+
+            // Collect the signature components
+            let { r, s, v } = await prepareDataSignatureParameters(
+              operator,
+              customTransactionType,
+              "MetaTransaction",
+              message,
+              metaTransactionsHandler.address
+            );
+
+            // Execute meta transaction, expecting revert.
+            await expect(
+              metaTransactionsHandler.executeMetaTransaction(
+                operator.address,
+                message.functionName,
+                functionSignature,
+                nonce,
+                r,
+                s,
+                v
+              )
+            ).to.revertedWith(RevertReasons.FUNCTION_NOT_WHITELISTED);
+          });
+
+          it("Should fail when function name is not whitelised - incorrect name", async function () {
+            let incorrectFunctionName = "createSeller"; // function with this name does not exist (argument types are missing)
+
+            // Prepare the function signature for the facet function.
+            functionSignature = accountHandler.interface.encodeFunctionData("createSeller", [
+              seller,
+              emptyAuthToken,
+              voucherInitValues,
+            ]);
+
+            // Prepare the message
+            message.functionName = incorrectFunctionName;
+            message.functionSignature = functionSignature;
+
+            // Collect the signature components
+            let { r, s, v } = await prepareDataSignatureParameters(
+              operator,
+              customTransactionType,
+              "MetaTransaction",
+              message,
+              metaTransactionsHandler.address
+            );
+
+            // Execute meta transaction, expecting revert.
+            await expect(
+              metaTransactionsHandler.executeMetaTransaction(
+                operator.address,
+                message.functionName,
+                functionSignature,
+                nonce,
+                r,
+                s,
+                v
+              )
+            ).to.revertedWith(RevertReasons.FUNCTION_NOT_WHITELISTED);
+          });
+
           it("Should fail when function name is incorrect", async function () {
-            let incorrectFunctionName = "createSeller"; // there are no function argument types here.
+            let incorrectFunctionName = "redeemVoucher(uint256)"; // function name is whitelisted, but different than what we encode in next step
```

Comment:
Is this test case sufficient to test the collision you pointed out in your document between redeemVoucher and cancelVoucher?

Reply:
I added a test that test actual collision case